lib/mixlib/shellout/windows/core_ext.rb

```diff
@@ -171,10 +179,19 @@ def create3(args)
       end
 
       env = nil
+      env_list = []
+
+      # Retrieve the environment variables for the specified user.
+      if hash["with_logon"]
+        logon, passwd, domain = format_creds_from_hash(hash)
+        logon_type = hash["elevated"] ? LOGON32_LOGON_BATCH : LOGON32_LOGON_INTERACTIVE
+        token = logon_user(logon, domain, passwd, logon_type)
+        env_list = retrieve_environment_variables(token, env_list)
+      end
 
       # The env string should be passed as a string of ';' separated paths.
       if hash["environment"]
-        env = hash["environment"]
+        env =  env_list.any? ? env_list : hash["environment"]
```

Comment:
env_list being an array and the comment above about it being a string don't seem to match